src/script/sscsm_controller.h

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2024 Luanti authors
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <memory>
+#include "irrlichttypes.h"
+#include "sscsm_irequest.h"
+#include "sscsm_ievent.h"
+#include "util/basic_macros.h"
+
+class SSCSMEnvironment;
+class StupidChannel;
+
+class SSCSMController
```

Comment:
can we name this something else?

Reply:
Do you have something in mind?

Reply:
not really. here's a few I can come up with:
* RemoteController
* IPCController
* just CSMController since "CPCSM" was never intended anyway
* ...

Reply:
For clarification, the purpose of this class is to:
* Be the RAII owner of the SSCSM process.
* Send events to SSCSM process, and process requests. (`runEvent`)
* Hide details (e.g. that it is a separate process, or that it has to do IPC calls).

So I think `SSCSMController` is a suitable name.

>RemoteController

The SSCSM process is not remote.

>IPCController

It's not about IPC, but about the SSCSM stuff. The IPC thing is internal.

>just CSMController since "CPCSM" was never intended anyway

CPCSM currently takes the name CSM / client scripting everywhere (we should rename it imo). So this would in practice be a name clash, and cause confusion.

Reply:
> For clarification, the purpose of this class is to:
> 
> Be the RAII owner of the SSCSM process.
> Send events to SSCSM process, and process requests. (runEvent)
> Hide details (e.g. that it is a separate process, or that it has to do IPC calls).

Can something like this be a doc comment of the class?

For lack of a better suggestion, I think going with `SSCSMController` is fine. If a better suggestion pops up, a rename is easy enough to do later on.